File: hydrograph.engine/hydrograph.engine.cascading/src/main/java/hydrograph/engine/flow/utils/DefaultPluginManipulation.java
package hydrograph.engine.flow.utils;

import hydrograph.engine.batchbreak.plugin.BatchBreakPlugin;
import hydrograph.engine.core.flowmanipulation.FlowManipulationHandler;

import java.util.LinkedList;
import java.util.List;

/**
 * Created by gurdits on 12/15/2016.
 */
public class DefaultPluginManipulation extends FlowManipulationHandler {

    enum Plugins {
        batchBreak {
            @Override
            public String getPluginName() {
                return BatchBreakPlugin.class.getName();
            }
        };
        public abstract String getPluginName();
    }

    @Override
    public List<String> addDefaultPlugins() {
        List<String> registerdPlugins = new LinkedList<String>();
        for (Plugins plugin : Plugins.values())
            registerdPlugins.add(plugin.getPluginName());
        return registerdPlugins;
    }
}

File: hydrograph.engine/hydrograph.engine.spark/src/main/scala/hydrograph/engine/spark/batchbreak/plugin/DefaultPluginManipulation.scala
package hydrograph.engine.spark.batchbreak.plugin

import java.util

import hydrograph.engine.core.flowmanipulation.FlowManipulationHandler

/**
  * Created by gurdits on 12/15/2016.
  */
class DefaultPluginManipulation extends FlowManipulationHandler{
  override def addDefaultPlugins(): util.List[String] = new util.LinkedList[String]()

}
